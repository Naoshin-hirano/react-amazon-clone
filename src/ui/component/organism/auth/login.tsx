import React from "react";
import "../../../../design/component/atoms/auth/Login.css";
import { Link, MemoryRouter } from "react-router-dom";

export const Login: React.FC<any> = (props) => {
    const { firstTitle } = props;
    console.log(firstTitle);

    return (
        <div className="login">
            <MemoryRouter>
                <Link to="/">
                    <img
                        className="login__logo"
                        src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
                    />
                </Link>
            </MemoryRouter>

            <div className="login__container">
                <h1>Sign In</h1>
                <form>
                    <h5>Email</h5>
                    <input type="text" />

                    <h5>Password</h5>
                    <input type="password" />

                    <button className="login_signInButton">Sign In</button>
                </form>

                <p>
                    By signing in you agree to Amazon FAKE CLONE's Conditions of
                    Use & Sale. Please see our Privacy Notice, or Cookies Notice
                    and our Interest-Based Ads Notice.
                </p>

                <button className="login__registerButton">
                    Create your Amazon Account
                </button>
            </div>
        </div>
    );
};
