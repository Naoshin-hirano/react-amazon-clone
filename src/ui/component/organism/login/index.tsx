import React from "react";
import "../../../../design/component/atoms/auth/login.css";
import { Link } from "@tanstack/react-location";

export const Login: React.FC<any> = (props) => {
    const { firstTitle } = props;
    console.log(firstTitle);

    return (
        <div className="login">
            <Link to="/">
                <img
                    alt="Amazon_logo"
                    className="login__logo"
                    src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
                />
            </Link>
            <div className="login__container">
                <h1>ログイン</h1>
                <form>
                    <h5>メール</h5>
                    <input type="text" />

                    <h5>パスワード</h5>
                    <input type="password" />

                    <button className="login_signInButton">ログイン</button>
                </form>

                <p>
                    サインインすることにより、Amazon FAKE CLONE
                    の条件に同意したことになります
                    使用と販売。当社のプライバシーに関する通知または Cookie
                    に関する通知をご覧ください。 およびインタレスト
                    ベース広告の通知。
                </p>

                <button className="login__registerButton">
                    Amazon Accountを作成する
                </button>
            </div>
        </div>
    );
};
