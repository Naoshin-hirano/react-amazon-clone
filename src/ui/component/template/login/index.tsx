import React from "react";
import { Login } from "ui/component/organism/auth/login";

export const Template: React.FC<any> = (props) => {
    return (
        <div>
            <Login {...props} />
        </div>
    );
};
