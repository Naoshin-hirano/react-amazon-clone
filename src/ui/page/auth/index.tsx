import React from "react";
import { Template } from "ui/component/template/auth";
import { useCreateProps } from "./presentation";

export const AuthPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
