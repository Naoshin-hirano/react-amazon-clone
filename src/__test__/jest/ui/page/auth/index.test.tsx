import { render } from "@testing-library/react";
import React from "react";
import { Login } from "ui/component/organism/auth/login";

describe("Login", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    // propsのモック
    const LoginPropsMock = {
        firstTitle: {
            title: "Firstタイトル",
        },
        firstMenu: {
            buttonLabels: {
                menu1: "Firstメニュー1",
                menu2: "Firstメニュー2",
                menu3: "Firstメニュー3",
            },
        },
        backButton: {
            buttonLabels: {
                back: "閉じる",
            },
        },
    };

    it("Login画面 snapshot確認", async () => {
        const { asFragment } = render(<Login {...LoginPropsMock} />);
        expect(asFragment()).toMatchSnapshot("Login");
    });
});
