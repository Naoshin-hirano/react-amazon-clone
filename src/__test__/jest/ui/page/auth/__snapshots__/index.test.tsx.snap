// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login画面 snapshot確認: Login 1`] = `
<DocumentFragment>
  <div
    class="login"
  >
    <a
      href="/"
    >
      <img
        class="login__logo"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Amazon_logo.svg/1024px-Amazon_logo.svg.png"
      />
    </a>
    <div
      class="login__container"
    >
      <h1>
        Sign In
      </h1>
      <form>
        <h5>
          Email
        </h5>
        <input
          type="text"
        />
        <h5>
          Password
        </h5>
        <input
          type="password"
        />
        <button
          class="login_signInButton"
        >
          Sign In
        </button>
      </form>
      <p>
        By signing in you agree to Amazon FAKE CLONE's Conditions of Use & Sale. Please see our Privacy Notice, or Cookies Notice and our Interest-Based Ads Notice.
      </p>
      <button
        class="login__registerButton"
      >
        Create your Amazon Account
      </button>
    </div>
  </div>
</DocumentFragment>
`;
